projects/memapCore/src/main/java/memap/helper/milp/MILPProblemNoConnections.java

```diff
@@ -342,12 +345,29 @@ public LpSolve createObjectiveFunction(LpSolve problem, BuildingMessage building
 				
 				if (topologyController.getOptimizationCriteria() == OptimizationCriteria.EUR) {
             		row[counter++] = pm.operationalCostEUR;
+//            		System.out.println("CP-costs " + pm.operationalCostEUR + " written at counter " + (counter-1));
 				}
             	
             	if (topologyController.getOptimizationCriteria() == OptimizationCriteria.CO2) {
             		row[counter++] = pm.operationalCostCO2;
             	}
             	
+            	if (topologyController.getToolUsage() == ToolUsage.SERVER && pm.varOperationalCostEUR != null) {
+            		
+            		// This part overwrites the previous costs if above condition is given
+            		counter--;
+            		// TODO: solve better
+            		
+            		if (topologyController.getOptimizationCriteria() == OptimizationCriteria.EUR) {
+                		row[counter++] = pm.varOperationalCostEUR[i];
+//                		System.out.println("CP-costs " + pm.varOperationalCostEUR[i] + " overwritten at counter " + (counter-1));
+    				}
+                	
+                	if (topologyController.getOptimizationCriteria() == OptimizationCriteria.CO2) {
+                		row[counter++] = pm.varOperationalCostEUR[i];
+                	}	
+            	}
```

Comment:
We should try to harmonize it - maybe later - so that we have only one version for the server and the tool. I see the benefits of having variational costs for the component, though.